In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
from matplotlib import pyplot as plt


In [2]:
df=pd.read_excel(io='sales_rep.xlsx',sheet_name='sales_data_sample')
df2=pd.read_excel(io='sales_rep.xlsx',sheet_name='Sales_rep',)

In [3]:
df.head(2)

SALES_REP_ID  ORDERNUMBER  QUANTITYORDERED  PRICEEACH  ORDERLINENUMBER  \
0             1        10223               37      100.0                1   
1             1        10309               41      100.0                5   

     SALES        ORDERDATE   STATUS  PRODUCTLINE  MSRP PRODUCTCODE  \
0  3965.66   2/20/2004 0:00  Shipped  Motorcycles    95    S10_1678   
1  4394.38  10/15/2004 0:00  Shipped  Motorcycles    95    S10_1678   

                 CUSTOMERNAME       CITY POSTALCODE    COUNTRY DEALSIZE  
0  Australian Collectors, Co.  Melbourne       3004  Australia   Medium  
1          Baane Mini Imports    Stavern       4110     Norway   Medium

In [4]:
df2.head(2)

SALES_REP_ID               Name First Name  Last Name Gender  \
0             1  Harmani Calderoni    Harmani  Calderoni      M   
1             2   Singleton Ashton  Singleton     Ashton      M   

                                           Photo Img  
0  https://images.pexels.com/photos/91227/pexels-...  
1  https://images.pexels.com/photos/718261/pexels...

In [5]:
df=df.iloc[:,1:].merge(df2.iloc[:,:2],left_on='ORDERLINENUMBER',right_on='SALES_REP_ID')

In [6]:
df.head(10)

ORDERNUMBER  QUANTITYORDERED  PRICEEACH  ORDERLINENUMBER    SALES  \
0        10223               37     100.00                1  3965.66   
1        10318               46      94.74                1  4358.04   
2        10210               34     100.00                1  6123.40   
3        10377               50     100.00                1  5182.00   
4        10419               37     100.00                1  5202.94   
5        10405               55     100.00                1  8289.05   
6        10392               36     100.00                1  4035.96   
7        10240               37     100.00                1  5959.22   
8        10105               50      79.67                1  3983.50   
9        10317               35      83.32                1  2916.20   

             ORDERDATE   STATUS   PRODUCTLINE  MSRP PRODUCTCODE  \
0       2/20/2004 0:00  Shipped   Motorcycles    95    S10_1678   
1  2004-02-11 00:00:00  Shipped   Motorcycles    95    S10_1678   
2  2004-12-01 00:00:00  Shipped   Motorcycles   193    S10_4698   
3  2005-09-02 00:00:00  Shipped  Classic Cars   115    S12_4675   
4       5/17/2005 0:00  Shipped  Classic Cars   124    S18_1589   
5       4/14/2005 0:00  Shipped  Classic Cars   169    S18_3232   
6  2005-10-03 00:00:00  Shipped  Vintage Cars    99    S18_3320   
7       4/13/2004 0:00  Shipped  Classic Cars   141    S18_3685   
8  2003-11-02 00:00:00  Shipped  Vintage Cars    83    S24_3816   
9  2004-02-11 00:00:00  Shipped        Planes    72    S24_4278   

                 CUSTOMERNAME        CITY POSTALCODE    COUNTRY DEALSIZE  \
0  Australian Collectors, Co.   Melbourne       3004  Australia   Medium   
1       Diecast Classics Inc.   Allentown      70267        USA   Medium   
2        Osaka Souveniers Co.       Osaka   530-0003      Japan   Medium   
3        Toys of Finland, Co.    Helsinki      21240    Finland   Medium   
4       Salzburg Collectables    Salzburg       5020    Austria   Medium   
5                 Mini Caravy  Strasbourg      67000     France    Large   
6             Mini Auto Werke        Graz       8010    Austria   Medium   
7        Osaka Souveniers Co.       Osaka   530-0003      Japan   Medium   
8    Danish Wholesale Imports   Kobenhavn       1734    Denmark   Medium   
9        Technics Stores Inc.  Burlingame      94217        USA    Small   

   SALES_REP_ID               Name  
0             1  Harmani Calderoni  
1             1  Harmani Calderoni  
2             1  Harmani Calderoni  
3             1  Harmani Calderoni  
4             1  Harmani Calderoni  
5             1  Harmani Calderoni  
6             1  Harmani Calderoni  
7             1  Harmani Calderoni  
8             1  Harmani Calderoni  
9             1  Harmani Calderoni

In [7]:
print(df.shape)
df.info()

(2823, 17)
<class 'pandas.core.frame.DataFrame'>
Int64Index: 2823 entries, 0 to 2822
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   ORDERNUMBER      2823 non-null   int64  
 1   QUANTITYORDERED  2823 non-null   int64  
 2   PRICEEACH        2823 non-null   float64
 3   ORDERLINENUMBER  2823 non-null   int64  
 4   SALES            2823 non-null   float64
 5   ORDERDATE        2823 non-null   object 
 6   STATUS           2823 non-null   object 
 7   PRODUCTLINE      2823 non-null   object 
 8   MSRP             2823 non-null   int64  
 9   PRODUCTCODE      2823 non-null   object 
 10  CUSTOMERNAME     2823 non-null   object 
 11  CITY             2823 non-null   object 
 12  POSTALCODE       2747 non-null   object 
 13  COUNTRY          2823 non-null   object 
 14  DEALSIZE         2823 non-null   object 
 15  SALES_REP_ID     2823 non-null   int64  
 16  Name             2823 non-null   object 
dtypes: 

### The Data Structure
The data comprise 2823 rows and 17 columns (features). 6 of these 17 columns (35.3%) are numeric variables

The DEALSIZE is the only ordered factor variable (Small->Medium->Large)

The rest of the variables are non-numeric and they are self-explanatory from their column's name

## Data Accessment
***

## Data Cleaning

In [20]:
x=[x.lower() for x in df.columns ]

In [23]:
# paste and edit output x
x=['order_number',
 'quantity_ordered',
 'price_each',
 'order_line_number',
 'sales',
 'order_date',
 'status',
 'product_line',
 'msrp',
 'product_code',
 'customer_name',
 'city',
 'postal_code',
 'country',
 'deal_size',
 'sales_rep_id',
 'name']
# replace data frame's columns with x
df.columns=x

In [25]:
df.head(2)

order_number  quantity_ordered  price_each  order_line_number    sales  \
0         10223                37      100.00                  1  3965.66   
1         10318                46       94.74                  1  4358.04   

            order_date   status product_line  msrp product_code  \
0       2/20/2004 0:00  Shipped  Motorcycles    95     S10_1678   
1  2004-02-11 00:00:00  Shipped  Motorcycles    95     S10_1678   

                customer_name       city postal_code    country deal_size  \
0  Australian Collectors, Co.  Melbourne        3004  Australia    Medium   
1       Diecast Classics Inc.  Allentown       70267        USA    Medium   

   sales_rep_id               name  
0             1  Harmani Calderoni  
1             1  Harmani Calderoni

In [35]:
print('The data frame conatins {} duplicate(s)'.format (df.duplicated().sum()))

The data frame conatins 0 duplicate(s)
